test/auth_test.go

```diff
@@ -173,6 +231,10 @@ func TestTokenAuth(t *testing.T) {
 	if err != nil {
 		t.Fatalf("Should have connected successfully: %v", err)
 	}
+	nc, err = nats.Connect("nats://localhost:8232", nats.TokenFunc(func() string { return secret }))
```

Comment:
Might be useful to create a separate test for `TokenFunc`. And test a `badtoken` as well maybe.

Reply:
also you are missing `nc.Close()`